src/seedu/addressbook/storage/StorageFile.java

```diff
@@ -85,29 +93,40 @@ public StorageFile(String filePath) throws InvalidStorageFilePathException {
     private static boolean isValidPath(Path filePath) {
         return filePath.toString().endsWith(".xml");
     }
+    
+    /**
+     * Check if the file from the given file path exist
+     * @return true if file exist and false otherwise
+     */
+    private boolean isFileExist() {
+        File file = path.toFile();
+        return file.exists();
+    }
 
     /**
      * Saves all data to this storage file.
      *
+     * @throws FileNotFoundException if the file does not exist.
      * @throws StorageOperationException if there were errors converting and/or storing data to file.
      */
-    public void save(AddressBook addressBook) throws StorageOperationException {
-
+    public void save(AddressBook addressBook) throws StorageOperationException, FileNotFoundException {
+        if (!isFileExist()) {
```

Comment:
Thanks for pointing out my mistake! Will think of a better way to improve the user experience! (: